presto-spark/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskCompiler.java

```diff
@@ -0,0 +1,337 @@
+            TaskManagerConfig taskManagerConfig,
+            NodeMemoryConfig nodeMemoryConfig,
+            NodeSpillConfig nodeSpillConfig)
+    {
+        this(
+                sessionPropertyManager,
+                sparkTaskDescriptorJsonCodec,
+                taskStatsJsonCodec,
+                notificationExecutor,
+                yieldExecutor,
+                localExecutionPlanner,
+                blockEncodingSerde,
+                requireNonNull(nodeMemoryConfig, "nodeMemoryConfig is null").getMaxQueryMemoryPerNode(),
+                requireNonNull(nodeMemoryConfig, "nodeMemoryConfig is null").getMaxQueryTotalMemoryPerNode(),
+                requireNonNull(nodeSpillConfig, "nodeSpillConfig is null").getMaxSpillPerNode(),
+                requireNonNull(taskManagerConfig, "taskManagerConfig is null").isPerOperatorCpuTimerEnabled(),
+                requireNonNull(taskManagerConfig, "taskManagerConfig is null").isTaskCpuTimerEnabled());
+    }
+
+    public PrestoSparkTaskCompiler(
+            SessionPropertyManager sessionPropertyManager,
+            JsonCodec<SparkTaskDescriptor> sparkTaskDescriptorJsonCodec,
+            JsonCodec<TaskStats> taskStatsJsonCodec,
+            Executor notificationExecutor,
+            ScheduledExecutorService yieldExecutor,
+            LocalExecutionPlanner localExecutionPlanner,
+            BlockEncodingSerde blockEncodingSerde,
+            DataSize maxUserMemory,
+            DataSize maxTotalMemory,
+            DataSize maxSpillMemory,
+            boolean perOperatorCpuTimerEnabled,
+            boolean cpuTimerEnabled)
+    {
+        this.sessionPropertyManager = requireNonNull(sessionPropertyManager, "sessionPropertyManager is null");
+        this.sparkTaskDescriptorJsonCodec = requireNonNull(sparkTaskDescriptorJsonCodec, "sparkTaskDescriptorJsonCodec is null");
+        this.taskStatsJsonCodec = requireNonNull(taskStatsJsonCodec, "taskStatsJsonCodec is null");
+        this.notificationExecutor = requireNonNull(notificationExecutor, "notificationExecutor is null");
+        this.yieldExecutor = requireNonNull(yieldExecutor, "yieldExecutor is null");
+        this.localExecutionPlanner = requireNonNull(localExecutionPlanner, "localExecutionPlanner is null");
+        this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
+        this.maxUserMemory = requireNonNull(maxUserMemory, "maxUserMemory is null");
+        this.maxTotalMemory = requireNonNull(maxTotalMemory, "maxTotalMemory is null");
+        this.maxSpillMemory = requireNonNull(maxSpillMemory, "maxSpillMemory is null");
+        this.perOperatorCpuTimerEnabled = perOperatorCpuTimerEnabled;
+        this.cpuTimerEnabled = cpuTimerEnabled;
+    }
+
+    public Iterator<Tuple2<Integer, byte[]>> compile(
+            int sparkTaskId,
+            byte[] serializedTaskDescriptor,
+            Map<String, Iterator<Tuple2<Integer, byte[]>>> inputs,
```

Comment:
Add a comment  about this parameter: 
-  key type (is the string representation for `PlanNodeId`
- The key in Tuple2 will be ignored -- since in our (key, value) pair, the key is just the partition id.

Reply:
Actually, as a TODO, maybe refactor it into a dedicated class? (Basically a class contains a `HashMap` -- poor man's way of doing `typedef`)

Reply:
Created  `PrestoSparkTaskInputs` wrapper